zkevm-circuits/src/state_circuit/state.rs

```diff
@@ -1996,4 +1996,132 @@ mod tests {
             Ok(())
         );
     }
+
+    // To run this benchmark, comment the `ignore` flag and run the following
+    // command:
+    // `RUSTFLAGS='-C target-cpu=native' cargo test --profile bench
+    // bench_state_circuit_prover -- --nocapture`
+    #[ignore]
+    #[test]
+    fn bench_state_circuit_prover() {
+        use ark_std::{end_timer, start_timer};
+        use halo2::transcript::{Blake2bRead, Blake2bWrite, Challenge255};
+        use halo2::{plonk::*, poly::commitment::Setup};
+        use pairing::bn256::Bn256;
+        use rand::SeedableRng;
+        use rand_xorshift::XorShiftRng;
+
+        const DEGREE: u32 = 22;
+        const MEMORY_ROWS_MAX: usize = 1 << (DEGREE - 2);
+        const MEMORY_ADDRESS_MAX: usize = 2000;
+        const STACK_ROWS_MAX: usize = 1 << (DEGREE - 2);
+        const STACK_ADDRESS_MAX: usize = 1023;
+        const STORAGE_ROWS_MAX: usize = 1 << (DEGREE - 2);
```

Comment:
@miha-stopar is it still correct to leave the ROWS_MAX as `1 << (DEGREE - 2)`? That was the only way to avoid a failure during proof generation due to out_of_bounds.

Reply:
@CPerezz k needs to be big enough such that 2^k > MEMORY_ROWS_MAX + STACK_ROWS_MAX + STORAGE_ROWS_MAX. In this case, it's MEMORY_ROWS_MAX + STACK_ROWS_MAX + STORAGE_ROWS_MAX = 3 * 2^(k-2) < 4 * 2^(k-2)= 2^k, so this should be ok.